src/herder/test/HerderTests.cpp

```diff
@@ -3857,6 +3894,10 @@ TEST_CASE("soroban txs accepted by the network",
         nodes[0]->getMetrics().NewCounter({"ledger", "apply", "success"});
     auto& txsFailed =
         nodes[0]->getMetrics().NewCounter({"ledger", "apply", "failure"});
+    auto& sorobanTxsSucceeded = nodes[0]->getMetrics().NewCounter(
```

Comment:
seems like there's a test failure in CI, any ideas what's going on there?

Reply:
I've made some progress into figuring out what is going on here. I forgot that the CI system merges `master` into the PR before running tests. When building the same merged branch the CI system uses I am able to reproduce a test failure in the `soroban txs accepted by the network` test. I'll rebase this branch on top of `master` and fix the issue.

Reply:
This turned out to be due to new costs for uploading wasms in protocol 21. I fixed it in 7fbac3c820f4673833b0c19e83b2fed81dfbd362

Reply:
thanks for fixing! how come core crashed instead of failing gracefully due to new costs though?

Reply:
I think it was just a confusing error message, not a core crash as I originally thought. The test was failing because [this `crankUntil` call](https://github.com/bboston7/stellar-core/blob/afcaba136d80585906c569af967c3372e3876ce3/src/herder/test/HerderTests.cpp#L3931) was timing out due to loadgen failing to reach [the required 50% success rate](https://github.com/bboston7/stellar-core/blob/afcaba136d80585906c569af967c3372e3876ce3/src/herder/test/HerderTests.cpp#L3926) due to the insufficient fees. When `crankUntil` times out, it throws an exception, which resulted in the stack trace we saw. Running the test locally produced a much more helpful error message.

I'm not sure why the CI run dumped the stack trace it did instead of the more useful error I got running the test locally. Maybe it came down to the different compiler or build options in the CI system that led to some optimizations that weakened the ability to produce a useful stack trace?